app/services/proxy_status.rb

```diff
@@ -0,0 +1,55 @@
+class ProxyStatus
+  CONNECTION_ERRORS = [Errno::EINVAL, Errno::ECONNRESET, EOFError, Timeout::Error, Errno::ENOENT,
+                       Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError]
+
+  def initialize(proxy, opts = {})
+    raise ::Foreman::Exception.new(N_('Must specify a Smart Proxy to use')) unless proxy.is_a?(SmartProxy)
+    @proxy = proxy
+    @cache_duration = opts[:cache_duration] || 3.minutes
+  end
+
+  def proxy_version_api
```

Comment:
No need to prefix methods with `proxy` within a ProxyStatus IMO, it makes it redundant e.g `proxy_status.proxy_version_api` vs `proxy_status.api_version` - ditto below on `versions`